lib/Optimizer/Transforms/DecompositionPatterns.cpp

```diff
@@ -154,6 +172,23 @@ class QuakeOperatorCreator {
     return op;
   }
 
+  template <typename OpTy>
+  OpTy create(Location location, Type &measOut, SmallVector<Type> &wires,
+              SmallVector<Value> &targets, StringAttr &registerNameAttr) {
+    OpTy op;
+    op = rewriter.create<OpTy>(location, measOut, wires, targets,
+                               registerNameAttr);
+
+    auto resultWires = op.getWires();
+    auto resultIt = resultWires.begin();
+    auto resultWiresEnd = resultWires.end();
+    for (auto &t : targets)
```

Comment:
NIT when iterating container whose name is a plural it's cleaner to use the singular form of the word fof the iterator

for(auto box : boxes) 
for(auto &target : targets) 
for(auto value : values)

only (obvious) exception is for words that have the same singlural and plural form....
for(auto oneSheep : sheep) //this is ugly

Reply:
I will keep it as it is for now to be conssistent with the other implementations so we can upstream it later